cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/testdata/gcloud-setup.md

```diff
@@ -0,0 +1,151 @@
+  --docker-server=us-west1-docker.pkg.dev \
+  --docker-username=oauth2accesstoken \
+  --docker-password="$(gcloud auth application-default print-access-token)"
+
+$ flux create source helm podinfo-9 \
+       --url=oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone \
+       --namespace=default \
+       --secret-ref=gcp-repo-auth-9
+
+$ flux create hr podinfo-9 \
+      --source=HelmRepository/podinfo-9.default \
+      --chart=podinfo
+✚ generating HelmRelease
+► applying HelmRelease
+✔ HelmRelease created
+◎ waiting for HelmRelease reconciliation
+✔ HelmRelease podinfo-9 is ready
+✔ applied revision 6.1.8
+```
+  * with JSON key file (long-lived)
+```
+kubectl create secret docker-registry gcp-repo-auth \
+  --docker-server=us-west1-docker.pkg.dev \
+  --docker-username=_json_key \
+  --docker-password="$(cat $GOOGLE_APPLICATION_CREDENTIALS)"
+
+$ flux create source helm podinfo-7 \
+       --url=oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone \
+       --namespace=default \
+       --secret-ref=gcp-repo-auth
+✚ generating HelmRepository source
+► applying HelmRepository source
+✔ source created
+◎ waiting for HelmRepository source reconciliation
+✔ HelmRepository source reconciliation completed
+
+$ flux create hr podinfo-7 \
+>      --source=HelmRepository/podinfo-7.default \
+>      --chart=podinfo
+✚ generating HelmRelease
+► applying HelmRelease
+✔ HelmRelease updated
+◎ waiting for HelmRelease reconciliation
+✔ HelmRelease podinfo-7 is ready
+✔ applied revision 6.1.8
+```
+
+## Notes
+1. After a cold start (several hours of not running the tests), running the test always first at first
+```
+$ go test -v -timeout 9999s -run TestKindClusterAvailablePackageEndpointsForOCI 
+...
+--- PASS: TestKindClusterAvailablePackageEndpointsForOCI/Testing_[oci://ghcr.io/gfichtenholt/stefanprodan-podinfo-clone]_with_basic_auth_secret (18.25s)
+--- PASS: TestKindClusterAvailablePackageEndpointsForOCI/Testing_[oci://ghcr.io/gfichtenholt/stefanprodan-podinfo-clone]_with_dockerconfigjson_secret (16.72s)
+--- PASS: TestKindClusterAvailablePackageEndpointsForOCI/Testing_[oci://demo.goharbor.io/stefanprodan-podinfo-clone]_with_basic_auth_secret (17.51s)
+--- FAIL: TestKindClusterAvailablePackageEndpointsForOCI/Testing_[oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone]_with_service_access_token (5.51s)
+```
+
+due to this error on server:
+```
+I0825 06:01:17.145356       1 docker_reg_v2_repo_lister.go:50] ORAS v2 Registry [oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone PlainHTTP=false] PING: GET "https://us-west1-docker.pkg.dev/v2/": unexpected status code 401: unauthorized: No valid credential was supplied.
+```
+**TODO** The workaround is to re-start kubeapps-internal-kubeappsapis pod. So there appears to be some stale state left on a pod. I need to find what it is and how to properly fix it
```

Comment:
That is strange, at least there is a workaround. Do we need a specific GH issue for this?

Reply:
Its a GCP (google cloud platform) issue, not github. I am currently investigating it. I have some ideas, hopefully the fix will be in next PR